include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,343 @@
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "sam" },
+    };
+
+    //!\copydoc sequence_file_input_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                     & stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                        & sequence,
+              id_type                         & id,
+              qual_type                       & qualities)
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+
+        auto stream_it = ranges::begin(stream_view);
+
+        // cache the begin position so we write quals to the same position as seq in seq_qual case
+        size_t sequence_size_before = 0;
+        size_t sequence_size_after = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size_before = ranges::size(sequence);
+
+        auto constexpr is_tab = is_char<'\t'>;
+        auto constexpr is_free = is_char<'\r'>;
+
+        while (is_char<'@'>(*stream_it))
+        {
+            detail::consume(stream_view | view::take_line);
+            stream_it = ranges::begin(stream_view);
+        }
+
+        /* ID */
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            if (options.truncate_ids)
+            {
+              ranges::copy(stream_view | view::take_until(is_blank)
+                                       | view::char_to<value_type_t<id_type>>,
+                                   ranges::back_insert_iterator{id});
+              detail::consume(stream_view | ranges::view::take_while(is_tab));
+              detail::consume(stream_view | ranges::view::take_while(!is_tab));
+            }
+            else
+            {
+              ranges::copy(stream_view | view::take_until_or_throw(is_tab)
+                                       | view::char_to<value_type_t<id_type>>,
+                                   ranges::back_insert_iterator{id});
+            }
+
+        }
+        else
+        {
+             detail::consume(stream_view | ranges::view::take_while(!is_tab));
+        }
+
+          // Jump over all parts we are not interested in
+        for (int i = 0; i < 8;i++)
+        {
+            ++stream_it;
```

Comment:
Can you check if deleting this line changes anything? because I think it does not. The stream_iterator is invalidated as soon as you call `detail::consume`

Reply:
It changes things, because then the tests fail and in my opinion that makes sense, the ++stream_it jumps over the tab, which is not consumed by detail::consume(stream_view | ranges::view::take_while(!is_tab));